### PR TITLE
Update return types of Fhirstore methods to match FHIR spec

### PR DESCRIPTION
**Pull request by simonvadee**

### CODE REVIEW

File: fhirstore/fhirstore.py [hunk 1]
```diff
@@ -14,20 +14,21 @@
...
+    BadRequestError is returned when the client request could not be processed.
     """
 
     pass
 
 
-class BadRequestError(Exception):
+class NotSupportedError(Exception):
     """
     BadRequestError is returned when the client request could not be processed.
```
Comment (Jasopaum) [inline fhirstore/fhirstore.py]: You forgot to change the docstring

File: fhirstore/fhirstore.py [hunk 2]
```diff
@@ -122,13 +121,15 @@ def format_error(self, errors: List[str], severity="error", code="invalid") -> O
...
+    def create(
+        self, resource: Union[Dict, FHIRAbstractModel]
+    ) -> Union[FHIRAbstractModel, OperationOutcome]:
         """
         Creates a resource. The structure of the resource will be checked
         against its json-schema FHIR definition.
 
         Args:
-            - resource: either a dict with the resource data or a fhir.resources.FHIRAbstractModel
+            - resource: the created resource as fhir.resources.FHIRAbstractModel
```
Comment (Jasopaum) [inline fhirstore/fhirstore.py]: the resource to create?

File: fhirstore/fhirstore.py [hunk 3]
```diff
@@ -142,11 +143,21 @@ def create(self, resource: Union[Dict, FHIRAbstractModel]):
                     for err in e.errors()
                 ]
             )
+        except NotSupportedError as e:
+            return self.format_error([str(e)], code="not-supported")
```
Comment (Jasopaum) [inline fhirstore/fhirstore.py]: What do you think about having a `format` method on `NotSupportedError`, `BadRequestError`, etc.?
We could fill stuff like code, etc. inside and return e.format() here (and remove format_error)

Comment (simonvadee) [reply fhirstore/fhirstore.py]: good idea, I didn't like this format_error anyway

File: fhirstore/fhirstore.py (hunk 2 above)
Comment (simonvadee) [reply fhirstore/fhirstore.py]: yes

File: fhirstore/fhirstore.py (hunk 1 above)
Comment (simonvadee) [reply fhirstore/fhirstore.py]: thx

Comment (simonvadee) [reply fhirstore/fhirstore.py]: done

File: fhirstore/fhirstore.py (hunk 2 above)
Comment (simonvadee) [reply fhirstore/fhirstore.py]: done

File: fhirstore/fhirstore.py (hunk 3 above)
Comment (simonvadee) [reply fhirstore/fhirstore.py]: done (much better thanks)

**Review by Jasopaum — APPROVED** [review]

**Pull request merged**